finality-aleph/src/data_io.rs

```diff
@@ -96,14 +121,17 @@ where
                 }
                 Some(block) = &mut import_stream.next() => {
                     trace!(target: "afa", "Block import notification at Data Store for block {:?}", block);
-                    self.add_block(block.hash);
+                    self.add_block(AlephData {hash: block.hash, number: *block.header.number()});
```

Comment:
Actually now that I think about it, maybe we shouldn't also add some check to see if
```
self.client.header(BlockId::Hash(block_data.hash)).number() == block_data.number
```

Reply:
This is tricky, because this change would make it incorrect ;/
The reason is that for a pair `(hash, fake_num)` if you have the block `hash` then you will discard this message (this is what you mean?), if you don't have it, then you might let it through during maintenance. So the behavior of message discarding would not be consistent.